File: 0x08-python-more_classes/4-rectangle.py
#!/usr/bin/python3

"""This module defines a class 'Rectangle'
"""


class Rectangle:
    """A Rectangle class with the attributes 'width'
       and 'height'
    """
    def __init__(self, width=0, height=0):

        self.width = width
        self.height = height

    @property
    def width(self):
        return self.__width

    @width.setter
    def width(self, value):

        self.__width = value

        if type(value) is not int:
            raise TypeError("with must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")

    @property
    def height(self):
        return self.__height

    @height.setter
    def height(self, value):

        self.__height = value

        if type(value) is not int:
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")

    def area(self):
        return self.__width * self.__height

    def perimeter(self):
        return (self.__width*2) + (self.__height*2)

        if width or height == 0:
            return 0

    def __str__(self):
        string = ""

        if self.__width == 0 or self.__height == 0:
            return string

        for i in range(self.__height):
            string += ("#" * self.__width)
            if i < self.__height - 1:
                string += '\n'
        return string

    def __repr__(self):
        return "{self.__class__.__name__}({self.width}, {self.height})".\
                 format(self=self)

File: 0x13-javascript_objects_scopes_closures/4-rectangle.js
#!/usr/bin/node
// contains 'Rectangle' class that defines a rectangle

class Rectangle {
  constructor (w, h) {
    if (w > 0 && h > 0) {
      this.width = w;
      this.height = h;
    }
  }

  print () {
    let storeX = '';
    for (let i = 0; i < this.height; i++) {
      for (let j = 0; j < this.width; j++) {
        storeX += 'X';
      }
      console.log(storeX);
      storeX = '';
    }
  }

  rotate () {
    this.width = this.height;
    this.height = this.width;
  }

  double () {
    this.width *= 2;
    this.height *= 2;
  }
}
module.exports = Rectangle;
